data/src/main/kotlin/com/vultisig/wallet/data/utils/RpcExtensions.kt

```
@@ -0,0 +1,46 @@
+    } else {
+        throw NetworkException(status.value, bodyAsText())
+    }
+}
+
+// TODO: Enhance and check for key recursively
+suspend fun extractError(response: HttpResponse, errorKey: String): String {
+    return try {
+        val responseBody = response.bodyAsText()
+        val json = JSONObject(responseBody)
+        return json.optString(errorKey, responseBody)
+    } catch (t: Throwable) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)